app/models/ab_test_session.rb

```diff
@@ -0,0 +1,37 @@
+# reference document to map a user/metrics_uuid to a feature_flag A/B test group
+class AbTestSession
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  # use declarative labels so reports are clear what each group saw
+  GROUP_NAMES = %w[control experiment].freeze
+  belongs_to :feature_flag
+  belongs_to :user, optional: true, foreign_key: :metrics_uuid, primary_key: :metrics_uuid, dependent: :delete_all
+
+  field :metrics_uuid, type: String
+  field :group_name, type: String
+
+  validates :metrics_uuid, uniqueness: { scope: :feature_flag_id }, presence: true
+  validates :group_name, presence: true, inclusion: { in: GROUP_NAMES }
+  before_validation :assign_group, on: :create
+
+  def self.random_group
+    rand.round.zero? ? GROUP_NAMES.first : GROUP_NAMES.last
```

Comment:
I don't think it's necessarily in scope here, but it seems worth raising: how would this extend to _multiple_ interventions in a given A/B test?  

In other words, how extensible is this design for A/B/C/D... tests?  That's a foreseeable next step, and it seems worth contemplating (though not necessarily implementing) now in case obvious trivial changes here would avoid major re-work down the line.

Reply:
It would not, but the the refactoring to allow that would be fairly straightforward.  Also, lacking a clear directive to support it, I agree it's out of scope here.

But since you asked... (this is just a 5 minute attempt, it could be cleaner)
```
# divide group names into n ranges
GROUP_NAMES = %w[control intervention combined neither].freeze
interval = 1.0 / GROUP_NAMES.size
ranges = 0.step(1 - interval, interval).map { |i| (i..i + interval) }
GROUPS_BY_RANGE = Hash[ranges.zip(GROUP_NAMES)]
```

This gives you:
```
{0.0..0.25=>"control", 0.25..0.5=>"intervention", 0.5..0.75=>"combined", 0.75..1.0=>"neither"}
```

Then our random group assignment is:
```
def self.random_group
  assignment = rand
  GROUPS_BY_RANGE.select { |range, _| range === assignment }.values.first
end
```

Reply:
Nice!

Reply:
In thinking on this, one thing to note is that this won't be configurable _per test_.  Specifically, if we made this change, all future tests would be A/B/C/D (or however many groups we define).  If we think that we will have occasion to do multiple kinds concurrently, then we'd need to refactor this logic out to a separate class, or put it in `FeatureFlag`.  

Doing the latter might make the most sense - group names would be defined in the `FeatureFlag`, and then this class would read the values from the parent flag and use the above logic to do the assignment.  That would then allow us to do test of 2 or more than 2 concurrently.  The validations would change slightly but not irrevocably so.